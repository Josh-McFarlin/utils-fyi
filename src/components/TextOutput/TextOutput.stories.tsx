import * as React from "react";
import TextOutput, { PropTypes } from "./TextOutput";

export default {
  title: "TextOutput",
  component: TextOutput,
};

export const Component = (args: PropTypes) => <TextOutput {...args} />;
Component.args = {
  children: "Some output goes here!",
};
