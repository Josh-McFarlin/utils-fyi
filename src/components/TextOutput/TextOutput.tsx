import * as React from "react";
import clsx from "clsx";
import classes from "./TextOutput.module.scss";

interface Props extends React.HTMLProps<HTMLInputElement> {
  className?: string;
  outputClassName?: string;
  label?: string;
  children: string | number;
}

/**
 * An input that can be used for any type of data.
 */
const TextOutput = ({
  className,
  outputClassName,
  label,
  children,
  ...rest
}: Props) => (
  <label className={clsx(classes.root, className)}>
    {label != null && <p>{label}</p>}
    <input className={outputClassName} value={children} readOnly {...rest} />
  </label>
);

export default TextOutput;
