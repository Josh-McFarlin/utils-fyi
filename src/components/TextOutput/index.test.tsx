import React from "react";
import { render } from "../../../test/testUtils";
import TextOutput from "./TextOutput";

describe("Button Component", () => {
  it("matches snapshot", () => {
    const { asFragment } = render(
      <TextOutput>Some output goes here!</TextOutput>,
      {}
    );

    expect(asFragment()).toMatchSnapshot();
  });
});
