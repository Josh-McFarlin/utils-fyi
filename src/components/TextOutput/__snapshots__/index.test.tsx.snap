// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button Component matches snapshot 1`] = `
<DocumentFragment>
  <label
    class="root"
  >
    <input
      readonly=""
      value="Some output goes here!"
    />
  </label>
</DocumentFragment>
`;
